### PR TITLE
Remove guava from core

### PR DESCRIPTION
**Pull request by fireuse**

### CODE REVIEW

**Comment by schildbach** [comment]
Also see #2000 and #2001.

**Comment by fireuse** [comment]
@schildbach Could you possibly review changes in the last commit?

File: core/build.gradle [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies {
     testImplementation 'org.hamcrest:hamcrest-library:1.3'
 }
 
-sourceCompatibility = 1.7
+sourceCompatibility = 1.8
```
Comment (schildbach) [inline core/build.gradle:28]: At the moment, we're still targeting JDK 7/Android 6.0. We have talked about raising this to something like JDK 8/Android 7.0 for version 0.17, but it's not decided yet.

File: core/src/main/java/org/bitcoinj/core/Bech32.java [hunk 2]
```diff
@@ -103,10 +101,18 @@ public static String encode(final Bech32Data bech32) {
         return encode(bech32.hrp, bech32.data);
     }
 
+    private static void checkHRPLen(String hrp){
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/Bech32.java:104]: I'm not convinced this is preferable to helpers like `checkArgument()`. Is there no JDK equivalent available?

File: core/src/main/java/org/bitcoinj/core/Block.java [hunk 3]
```diff
@@ -17,21 +17,21 @@
 
 package org.bitcoinj.core;
 
-import com.google.common.annotations.*;
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/Block.java:20]: Please do not reorder imports if the topic of your PR is actually something else.

File: core/src/main/java/org/bitcoinj/core/Block.java [hunk 4]
```diff
@@ -470,8 +469,8 @@ public String toString() {
         s.append(" block: \n");
         s.append("   hash: ").append(getHashAsString()).append('\n');
         s.append("   version: ").append(version);
-        String bips = Joiner.on(", ").skipNulls().join(isBIP34() ? "BIP34" : null, isBIP66() ? "BIP66" : null,
-                isBIP65() ? "BIP65" : null);
+        String bips = ", " + (isBIP34() ? "BIP34" : null) + (isBIP66() ? "BIP66" : null) +
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/Block.java:472]: This is not equivalent! See the documentation of `skipNulls()`.

File: core/src/main/java/org/bitcoinj/core/BlockLocator.java [hunk 5]
```diff
@@ -18,30 +18,35 @@
...
     }
 
     /**
      * Add a {@link Sha256Hash} to a newly created block locator.
      */
-    public BlockLocator add(Sha256Hash hash) {
-        return new BlockLocator(new ImmutableList.Builder<Sha256Hash>().addAll(this.hashes).add(hash).build());
+    public BlockLocator add(final Sha256Hash hash) {
+        return new BlockLocator(new LinkedList<Sha256Hash>(){{
+            addAll(hashes);
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/BlockLocator.java:47]: This is scary, a static block in an anonymous class…

**Review by schildbach — COMMENTED** [review]
Why do you want to "remove Guava from core"? I'm certainly not against replacing individual Guava idioms by their JDK counterpart, if there is one (this is usually indicated in the Guava JavaDoc). But why copy and paste Guava code?

In general for this kind of PR I'd prefer individual PRs (or at least individual commits) per kind of refactoring, rather than a commit per class. That way, each idea could be discussed and merged separately. Just as an example, your `ListenableFuture` to `CompletableFuture` refactoring would go into one dedicated commit.

**Comment by msgilligan** [comment]
I missed this one while away from the project temporarily. Nice to see the effort!

**Pull request closed without merging**